src/bin/cargo/commands/describe_future_incompatibilities.rs

```diff
@@ -0,0 +1,50 @@
+use crate::command_prelude::*;
+use anyhow::anyhow;
+use cargo::core::compiler::future_incompat::{OnDiskReport, FUTURE_INCOMPAT_FILE};
+use cargo::core::nightly_features_allowed;
+use cargo::drop_eprint;
+use std::io::Read;
+
+pub fn cli() -> App {
+    subcommand("describe-future-incompatibilities")
+        .arg(
+            opt(
+                "id",
+                "identifier of the report [generated by a Cargo command invocation",
+            )
+            .value_name("id")
+            .required(true),
+        )
+        .about("Reports any crates which will eventually stop compiling")
+}
+
+pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
+    if !nightly_features_allowed() {
+        return Err(anyhow!(
+            "`cargo describe-future-incompatibilities` can only be used on the nightly channel"
+        )
+        .into());
+    }
+
+    let ws = args.workspace(config)?;
+    let report_file = ws.target_dir().open_ro(
+        FUTURE_INCOMPAT_FILE,
+        ws.config(),
+        "Future incompatible report",
+    )?;
+
+    let mut file_contents = String::new();
+    report_file
+        .file()
+        .read_to_string(&mut file_contents)
+        .map_err(|e| anyhow!("Failed to read report: {:?}", e))?;
```

Comment:
Generally the preferred method for Cargo error messages to use chaining to add context.  The error message format usually starts with a lowercase letter (similar to rustc).  Also, it is preferred to avoid displaying debug-formatted things.  This can be used with `use cargo::util::CargoResultExt;`.

```suggestion
        .chain_err(|| "failed to read report")?;
```